lib/galaxy/jobs/__init__.py

```diff
@@ -192,6 +192,9 @@ def __parse_job_conf_xml(self, tree):
                             else:
                                 self.handlers[tag] = [id]
 
+        # Must define at least one handler to have a default.
+        if not self.handlers:
+            raise ValueError("Job configuration file defines no valid handler elements.")
```

Comment:
My thought was this was very low-level code we may want to rip out and reuse other places - the planemo of job configurations for instance. For this reason, I didn't want to create that inter-dependency unnecessarily. Also this exception will never be thrown from the API so the advantages of galaxy.exceptions are minimized. That said ``ValueError`` isn't quite right either - any compromises in the Python stdlib or do you feel strongly that we should use galaxy.exceptions.